File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

console.log(typeof pdf);
console.log(pdf);

let dataBuffer = fs.readFileSync('D:/bundles/ecommerce.pdf');

const parseFunc = typeof pdf === 'function' ? pdf : pdf.default || pdf.pdf;

parseFunc(dataBuffer).then(function(data) {
  console.log(data.text);
}).catch(function(error) {
  console.error('Error fetching PDF:', error);
});

File: read_pdf.py
import sys
import PyPDF2

try:
    with open('D:/bundles/ecommerce.pdf', 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        with open('parsed_pdf.txt', 'w', encoding='utf-8') as out:
            out.write(text)
except Exception as e:
    print(f"Error reading PDF: {e}")
